### ISSUE TITLE
java.lang.NoSuchMethodError: No virtual method onNestedFling

### ISSUE THREAD

**Issue by MiralDesai** [intercom/intercom-android#367]
This is different to the other NoSuchMethodError exception so I will open another issue.

## Version info
  - Intercom for Android version: 3.2.6
  - Android OS version: 6.1
  - Android Device: Nexus 5

## Expected behavior

For intercom not to crash my app.

## Actual behavior

I have an unread message, I open the inbox, I select the message and the app crashes.

## Steps to reproduce
 1. Send message
 2. Try to read message
 3. Blows up

## Logs

```
FATAL EXCEPTION: main
Process: com.cakeapp, PID: 15649
 java.lang.NoSuchMethodError: No virtual method onNestedFling(Landroid/support/design/widget/CoordinatorLayout;Landroid/support/design/widget/AppBarLayout;Landroid/view/View;FFZ)Z in class Landroid/support/design/widget/AppBarLayout$Behavior; or its super classes (declaration of 'android.support.design.widget.AppBarLayout$Behavior' appears in /data/app/com.myapp-2/base.apk:classes57.dex)
at io.intercom.android.sdk.profile.ProfilePresenter.flingProfile(ProfilePresenter.java:197)
at io.intercom.android.sdk.profile.ProfilePresenter.flingProfileOpen(ProfilePresenter.java:181)
 at io.intercom.android.sdk.profile.ProfilePresenter.profileAutoOpened(ProfilePresenter.java:172)
at io.intercom.android.sdk.conversation.ConversationFragment$8.run(ConversationFragment.java:657)
at android.os.Handler.handleCallback(Handler.java:739)
at android.os.Handler.dispatchMessage(Handler.java:95)
at android.os.Looper.loop(Looper.java:148)
 at android.app.ActivityThread.main(ActivityThread.java:5417)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
```

**Comment by ronocod** [comment]
What version of the support library are you using?

**Comment by MiralDesai** [comment]
26.0.0 beta 2, as part of the upgrade to Android O.

**Comment by ronocod** [comment]
It looks like that's the issue. Can you use version 25 for now?

**Comment by MiralDesai** [comment]
That would also mean downgrading my targetSdkVersion back to 25. API 26 (Android O) requires version 26 of the support library. 

I would appreciate if you could support Android O at least to the point of bug fixes, so that I can carry on with my development in that regard. I know it's going exactly going to be instant, but it would really help since my plan was to work on a lot of Android O related apis.

**Comment by MiralDesai** [comment]
Can I get a timeline on when this might be worked on? At least so I can plan around it.

Thanks.

**Comment by ronocod** [comment]
We won't be releasing version of the SDK using Support Library 26 before the final release comes out. I'll look into this to see if there's anything we can do to help in the meantime.

**Comment by MiralDesai** [comment]
Can I suggest a beta version of the sdk be made available with support? With full knowledge that things might break. I'm happy to keep it unreleased, and only have the beta version so that I make progress development wise.

I can understand not releasing anything until the support library is final, I certainly won't be either. A beta version might be a good idea for future breaking changes too.

Anyway just an idea.

**Comment by lpredova** [comment]
Anything new regarding this issue? 26.0.0 is not in alpha/beta anymore 😃 

https://developer.android.com/topic/libraries/support-library/revisions.html

**Comment by Bringoff** [comment]
This happens to me using library version `26.0.0` which is not beta anymore. No need to send messages to reproduce it, crash happens after opening any dialog, even already read.

**Comment by Bringoff** [comment]
This happens to me using intercom 3.2.10 and support library version 26.0.0 which is not beta anymore. No need to send messages to reproduce it, crash happens after opening any dialog, even already read.

**Comment by DaleCantwell** [comment]
This is fixed in [4.0.0](https://github.com/intercom/intercom-android/releases/tag/4.0.0)

**Comment by MiralDesai** [comment]
Woo!
